core/src/main/codegen/templates/Parser.jj

```diff
@@ -1988,8 +1988,8 @@ SqlLiteral JoinType() :
 }
 {
     (
-    LOOKAHEAD(3) // required for "LEFT SEMI JOIN" in Babel
```

Comment:
Why move it?

Reply:
In fact, I also have doubts about this `LOOKAHEAD(3)`, because `LEFT ANTI JOIN` also requires it, otherwise it will recognize errors, but `LOOKAHEAD(3)` is outside `<#list>`, so `joinType = ${method} ()` will generate  `LOOKAHEAD(2)` except for the first one `LEFT SEMI JOIN`. Give me some time to think about it and  reply to you as soon as possible

Reply:
You can execute `javacc core/build/fmpp/fmppMain/javacc/Parser.jj`.

The result should be
```
$ javacc core/build/fmpp/fmppMain/javacc/Parser.jj
Java Compiler Compiler Version 7.0.9 (Parser Generator)
(type "javacc" with no arguments for help)
Reading from file core/build/fmpp/fmppMain/javacc/Parser.jj . . .
Note: UNICODE_INPUT option is specified. Please make sure you create the parser/lexer using a Reader with the correct character encoding.
File "TokenMgrError.java" does not exist.  Will create one.
File "ParseException.java" does not exist.  Will create one.
File "Token.java" does not exist.  Will create one.
File "SimpleCharStream.java" does not exist.  Will create one.
Parser generated successfully.
```
If you find warning, you should solve it.

Reply:
ok, thanks for your suggestion. I'll check it

Reply:
@JiajunBernoulli I just execute `javacc core/build/fmpp/fmppMain/javacc/Parser.jj`. 
The execution results are as expected : 
```shell
Java Compiler Compiler Version 7.0.10 (Parser Generator)
(type "javacc" with no arguments for help)
Reading from file core/build/fmpp/fmppMain/javacc/Parser.jj . . .
Note: UNICODE_INPUT option is specified. Please make sure you create the parser/lexer using a Reader with the correct character encoding.
File "TokenMgrError.java" does not exist.  Will create one.
File "ParseException.java" does not exist.  Will create one.
File "Token.java" does not exist.  Will create one.
File "SimpleCharStream.java" does not exist.  Will create one.
Parser generated successfully.
```
There is no warning